DynamicAnalyzer/views/android.py

```diff
@@ -41,6 +42,186 @@
 def key(d, key_name):
     return d.get(key_name)
 
+def stopAVD(adb):
+    print "\n[INFO] Stopping MobSF Emulator"
+    try:
+        # adb -s emulator-xxxx emu kill
+        args = [adb, '-s', getIdentifier(), 'emu', 'kill']
+        subprocess.call(args)
+    except:
+        PrintException("[ERROR] Stopping MobSF Emulator")
+
+
+def deleteAVD(avd_path, avd_name):
+    print "\n[INFO] Deleting emulator files"
+    try:
+        config_file = os.path.join(avd_path, avd_name + '.ini')
+        if os.path.exists(config_file):
+            os.remove(config_file)
+
+        # TODO: Sometimes there is an error here because of the locks that avd does - check this out
```

Comment:
Yeah, If there is an issue with the emulator closing (adb emu kill) and it is not closed properly (like if the user will ahut down the emulator by himself with taskmanager or something..) the avd.lock file will still be present and you won't able to delete it. If the user won't interact with the processes manually it shoud not happen.